concent_api/core/validation.py

```diff
@@ -50,7 +52,8 @@ def validate_id_value(value, field_name):
             error_code=ErrorCode.MESSAGE_VALUE_WRONG_LENGTH,
         )
 
-    if not value.isalnum():
+    valid_id_regex = re.compile(r'^[a-zA-Z0-9_-]*')
```

Comment:
Please move this to constants. There is no need to compile this on each validation.

Reply:
Done